### PR TITLE
Add previous, continue, and review buttons

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Add buttons/links for navigation and submission to the block edit page. Also adds the "X of Y" block progress text

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

#### New Features

- [x] Conditioned new functionality on a [FeatureFlag](https://github.com/civiform/civiform/wiki/Feature-Flags)
- [x] Wrote browser tests with the feature flag off and on, etc.

### CODE REVIEW

**Comment by greg-hyde** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/civiform/civiform/6715?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#6715** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6715?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6710** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6710?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6703** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6703?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6702** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6702?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6701** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6701?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6700** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6700?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6699** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6699?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6513** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6513?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 3 other dependent PRs ([#6514](https://github.com/civiform/civiform/pull/6514) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6514?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#6531](https://github.com/civiform/civiform/pull/6531) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6531?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#6581](https://github.com/civiform/civiform/pull/6581) <a href="https://app.graphite.dev/github/pr/civiform/civiform/6581?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/6715?utm_source=stack-comment).

File: server/app/views/applicant/NorthStarApplicantProgramBlockEditView.java [hunk 1]
```diff
@@ -41,4 +43,19 @@ private String getFormAction(ApplicationBaseView.Params params) {
             ApplicantRequestedAction.NEXT_BLOCK)
         .url();
   }
+
+  private String getReviewUrl(ApplicationBaseView.Params params) {
+    return applicantRoutes.review(params.profile(), params.applicantId(), params.programId()).url();
+  }
```
Comment (caitlinshk) [inline server/app/views/applicant/NorthStarApplicantProgramBlockEditView.java:49]: Quick drive-by here: I'm currently updating the "Review" and "Previous" buttons to also save the user's answers, which means they should also be submit buttons and not an `<a>` tags. See https://github.com/civiform/civiform/pull/6544 and https://github.com/civiform/civiform/pull/6547. Can we have the buttons in the north star also save answers?

Comment (greg-hyde) [reply server/app/views/applicant/NorthStarApplicantProgramBlockEditView.java]: updated. good catch!
